ftw/publisher/core/adapters/configure.zcml

```diff
@@ -100,8 +100,28 @@
         <adapter
               for="ftw.shop.interfaces.IShopItem"
               provides="ftw.publisher.core.interfaces.IDataCollector"
-              factory="ftw.publisher.core.adapters.shop_items.ShopItemVariations"
-              name="shop_item_adapter" />
+              factory="ftw.publisher.core.adapters.ftw_shop.ShopItemVariations"
+              name="shop_item_variations_adapter" />
+        <adapter
+              for="ftw.shop.interfaces.IShopItem"
+              provides="ftw.publisher.core.interfaces.IDataCollector"
+              factory="ftw.publisher.core.adapters.ftw_shop.ShopCategorizableReferences"
+              name="shop_item_references_adapter" />
+        <adapter
+              for="ftw.shop.interfaces.IShopCategory"
+               provides="ftw.publisher.core.interfaces.IDataCollector"
+              factory="ftw.publisher.core.adapters.ftw_shop.ShopCategorizableReferences"
+              name="shop_category_references_adapter" />
+        <adapter
+              for="ftw.shop.interfaces.IShopItem"
+              provides="ftw.publisher.core.interfaces.IDataCollector"
+              factory="ftw.publisher.core.adapters.ftw_shop.ShopCategorizableRanks"
+              name="shop_item_ranks_adapter" />
+        <adapter
+              for="ftw.shop.interfaces.IShopCategory"
```

Comment:
I'm not sure, but I think it could be possible to use a class instead of an interface here.
By using `for="ftw.shop.content.categorizeable.Categorizeable` it would work for both, categories and items, and only one ZCML adapter statement would be necessary..